core/src/main/java/org/eclipse/hono/util/CredentialsObject.java

```diff
@@ -94,7 +94,7 @@ public CredentialsObject setProperty(final String name, final Object value) {
      */
     @JsonIgnore
     public String getDeviceId() {
-        return (String) getProperty(CredentialsConstants.FIELD_PAYLOAD_DEVICE_ID);
+        return getProperty(CredentialsConstants.FIELD_PAYLOAD_DEVICE_ID, String.class);
```

Comment:
same here: the CredentialsObject has been designed to represent the messages received from the Credentials API. Using it for the payload exchanged via the HTTP management API *might* work for now but will certainly require more and more special case handling once the two APIs divert even more. Therefore I would also suggest to use distinct value object classes for both APIs ...

Reply:
It's not used by design. For the purpose of HTTP API we created new value objects that matches the spec. Take a look at `CommonCredential`, `PasswordCredential`, etc in `org.eclipse.hono.service.management.credentials` package. If there's a specific case where they overlap we can take a look into that and refactor things.

Reply:
Then the changes to this class are not needed for the HTTP management API? If so, then I think they should go into a separate PR, right?